File: app/src/main/java/com/raylabs/doggie/utils/tab/TabTitleDelegate.java
package com.raylabs.doggie.utils.tab;

public final class TabTitleDelegate {
    private final String[] titles;

    public TabTitleDelegate(String[] titles) {
        if (titles == null) throw new IllegalArgumentException("titles cannot be null");
        this.titles = titles;
    }

    public String requireAt(int position) {
        if (position < 0 || position >= titles.length) {
            throw new IndexOutOfBoundsException("Invalid index: " + position + ", size: " + titles.length);
        }
        return titles[position];
    }

    public int size() {
        return titles.length;
    }
}

File: app/src/main/java/com/raylabs/doggie/utils/tab/TabTitleDelegate.kt
package com.raylabs.doggie.utils.tab

class TabTitleDelegate(titlesInput: Array<String>?) {

    private val titles: Array<String>

    init {
        requireNotNull(titlesInput) { "titles cannot be null" }
        titles = titlesInput
    }

    fun requireAt(position: Int): String {
        if (position !in titles.indices) {
            throw IndexOutOfBoundsException("Invalid index: $position, size: ${titles.size}")
        }
        return titles[position]
    }

    fun size(): Int = titles.size
}
